### ISSUE TITLE
(workaround) Windows: unf_ext requires Ruby version >= 2.0, < 2.6

### ISSUE THREAD

**Issue by z64** [meew0/discordrb#604]
The binaries release for `unf_ext`, specifically for `x86-mingw32`, is currently only tagged with support for `>= 2.0, < 2.6`, so `discordrb` cannot be trivially installed using Ruby 2.6.

https://rubygems.org/gems/unf_ext/versions/0.0.7.5-x86-mingw32

Current workarounds:

- **See swarley's comments below!** or..
- Use Ruby >= 2.0, < 2.6 (2.4+ recommended)
- Compile `unf_ext` yourself (https://github.com/knu/ruby-unf_ext)

`unf_ext` issue: https://github.com/knu/ruby-unf_ext/issues/48

**Comment by swarley** [comment]
# Update

There are presently three known work arounds for Windows users on 2.6

------------------------------
## Bundler

This is the easiest fix if you're familiar with bundler, simply using a Gemfile for your bot will automatically install discordrb using `unf_ext 0.0.7.2`, the last available version before windows specific version limiting.

## Globally install unf_ext

If you absolutely need to have a global installation of discordrb, install `unf_ext` via the command `gem install unf_ext -v0.0.7.2` before `gem install discordrb`

## Build from source

```bat
git clone https://github.com/knu/ruby-unf_ext.git
cd ruby-unf_ext

rem Ensure that you have the ridk available
ridk enable

ruby ext\unf_ext\extconf.rb && make && make install
```

**Comment by swarley** [comment]
This issue is now fixed with the release of a [2.6 compatible unf_ext](https://github.com/knu/ruby-unf_ext/commit/90fd38bfd32784bfc5f7a7d74d70da7707fa56a2)

**Comment by z64** [comment]
yeet
